src/components/Tokens/TokenDetails/types.ts

```diff
@@ -0,0 +1,19 @@
+export interface Swap {
+  timestamp: number
+  usdValue: number
+  input: SwapInOut
+  output: SwapInOut
+  maker: string
+  transactionHash: string
+}
```

Comment:
we should have generated graphql types for this

Reply:
My understanding after reviewing the info repos is that V1, V2, and V3 transactions all have different schemas that are currently being queried, and none of them has a raw format that's ready to display (i.e. they all require some pre-processing). With that in mind, I thought it would be cleanest to define my own interface and implement adapters that can be paired with the queries in order to prepare the data before rendering it. Ideally, this would all be handled by the backend, and we could directly use the schema they settle on, but until then, this option should suffice. What do you think?